### ISSUE TITLE
BUG: indices[0] = [0,-1] is out of bounds for _OneHotColumn

### ISSUE THREAD

**Issue by facaiy** [tensorflow/tensorflow#12583]
### Describe the problem

For `sparse_column_with_keys`, out-of-vocabulary feature values is -1 by default. However, -1 is invalid index when sparse_tensor is converted to tensor in `one_hot_column`.

### Source code / logs

code:

```python
    ids = fc.sparse_column_with_keys("ids", ["marlo", "omar", "stringer"])
    weighted_ids = fc.weighted_sparse_column(ids, "weights")
    one_hot = fc.one_hot_column(weighted_ids)
```

logs:

```python
  File "/Users/facai/Workshop/sina/Prometheus/prometheus/python/feature/column_test.py", line 373, in _feature_column_from_tensor_dict
    return layers.input_from_feature_columns(tensor_dicts, feature_columns)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/contrib/layers/python/layers/feature_column_ops.py", line 212, in input_from_feature_columns
    default_name='input_from_feature_columns')
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/contrib/layers/python/layers/feature_column_ops.py", line 143, in _input_from_feature_columns
    output_rank=output_rank))
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/contrib/layers/python/layers/feature_column.py", line 936, in _to_dnn_input_layer
    return sparse_ops.sparse_tensor_to_dense(weighted_column)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/ops/sparse_ops.py", line 845, in sparse_tensor_to_dense
    name=name)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/ops/sparse_ops.py", line 710, in sparse_to_dense
    name=name)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/ops/gen_sparse_ops.py", line 1094, in _sparse_to_dense
    validate_indices=validate_indices, name=name)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 2506, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/Users/facai/Library/anaconda3/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1269, in __init__
    self._traceback = _extract_stack()

InvalidArgumentError (see above for traceback): indices[0] = [0,-1] is out of bounds: need 0 <= index < [1,4]
         [[Node: input_from_feature_columns/col_tag_weighted_by_col_weight_one_hot/SparseToDense = SparseToDense[T=DT_FLOAT, Tindices=DT_INT64, validate_indices=true, _device="/job:localhost/replica:0/task:0/cpu:0"](input_from_feature_columns/col_tag_weighted_by_col_weight_one_hot/SparseMerge/SparseReorder, input_from_feature_columns/col_tag_weighted_by_col_weight_one_hot/SparseMerge/Identity, input_from_feature_columns/col_tag_weighted_by_col_weight_one_hot/SparseMerge/SparseReorder:1, input_from_feature_columns/col_tag_weighted_by_col_weight_one_hot/SparseToDense/default_value)]]
```

**Comment by andydavis1** [comment]
@jhseu please close this once #12584 and #12558 are merged...

**Comment by facaiy** [comment]
Hi, the out-of-bounds bug in `tf.contrib.layers.feature_column` is fixed in #12584, and `tf.feature_columns` is fixed in #12638. Please close this once they are both accepted and merged. Thanks.
